### ISSUE TITLE
Validate the number of optimizer runs

### ISSUE THREAD

**Issue by fvictorio** [NomicFoundation/hardhat#3488]
Continuation of https://github.com/NomicFoundation/hardhat/issues/1589.

The max number of optimizer runs is $2^{32}-1$. If you use a bigger number, the error is:

```
The "runs" setting must be an unsigned number.
Error HH600: Compilation failed
```

While this comes from solc, we should add some validation here and show a better error message.

**Comment by konarshankar07** [comment]
Hello @fvictorio 

Sorry, I'm new to this framework and probably a dumb question but do you know how we can reproduce this issue?

**Comment by fvictorio** [comment]
Yes, just create a new project with some contracts and use this as your config:

```js
module.exports = {
  solidity: {
    version: "0.8.17",
    settings: {
      optimizer: {
        enabled: true,
        runs: 2 ** 32,
      },
    },
  },
};
```

Then if you run `npx hardhat compile` you'll see this error:

```
The "runs" setting must be an unsigned number.
Error HH600: Compilation failed
```

That error comes from the compiler, but Hardhat should throw a validation error before even trying to compile.

I'm not super sure what's the best place to do this though. Maybe this needs a "post config resolution" phase, that we don't have yet (but we want to add, for example for https://github.com/NomicFoundation/hardhat/issues/3323). But that's a bigger change. If you feel like tackling that, I can give you some hints.

**Comment by fvictorio** [comment]
@konarshankar07 I opened #3500, if you want to give it a try. I think that what we need is:

- Add a new `function validateResolvedConfig(config: HardhatConfig)` function in `src/internal/core/config/config-validation.ts` and validate the number of runs in the solidity configuration. We should validate both the list of compilers and any overrides.
- Add a new `INVALID_RESOLVED_CONFIG` error (under the `GENERAL` category) in `src/internal/core/errors-list.ts`.
- Add proper tests in `test/internal/core/config/config-validation.ts`

I know this might be a lot if it's your first contribution, so if you'd prefer to help with something easier, let me know!

**Comment by konarshankar07** [comment]
Thanks @fvictorio I will first work on 3500 ticket and then come back on this
